src/conv-core/msgmgr.C

```diff
@@ -168,7 +169,9 @@ CmmTable CmmPup(pup_er p, CmmTable t, CmmPupMessageFn msgpup)
       int ntags, *tags;
       void *msg;
       pup_int(p, &ntags);
-      tags = (int*) malloc(ntags*sizeof(int));
+      CmiEnforce(ntags <= std::numeric_limits<int>::max());
+      tags = (int*) malloc((size_t)ntags*sizeof(int));
```

Comment:
Can this be a std::vector? (same for threads.C)

Reply:
This CmmTable code does highly suspicious things like allocating these tags as subregions of a single CmiAlloc when you create them, but separate mallocs when unpacking. It should probably be on the chopping block entirely. I don't want to touch it now, and I've removed the limit check anyway.

In threads.C, the pointer to the buffer that is allocated is the return value of the function.